crates/optimism/txpool/src/interop.rs

```diff
@@ -0,0 +1,41 @@
+//! Additional support for pooled transactions with [`TransactionInterop`]
+
+/// Helper trait that allows attaching a [`TransactionInterop`].
+pub trait MaybeInteropTransaction {
+    /// Attach a [`TransactionInterop`].
+    fn set_interop(&mut self, conditional: TransactionInterop);
```

Comment:
we can't take &mut here because the way txs are stored is always in an `Arc`.

and since we need to update this from time to time, we likely need to shared mut access here and should wrap it in an parking_lot::Rwlock

Reply:
i think we could, as https://github.com/paradigmxyz/reth/blob/fb5a6e755ddc8b9cc1c54ffc6deef775a2bf809a/crates/optimism/txpool/src/conditional.rs#L7 MaybeConditionalTransaction uses it the same way. 
But I agree that we need something for updates